internal/sqladapter/database.go

```diff
@@ -151,25 +151,23 @@ func NewBaseDatabase(p PartialDatabase) BaseDatabase {
 // BaseDatabase and PartialDatabase
 type database struct {
 	PartialDatabase
-	baseTx BaseTx
-
 	db.Settings
 
+	lookupNameOnce sync.Once
```

Comment:
use `sync.Once` to look up the DB name instead of the `mu` mutex -- this is what caused the DEADLOCK mentioned above